juniper_rocket/src/lib.rs

```diff
@@ -107,30 +107,46 @@ impl<'f> FromForm<'f> for GraphQLRequest {
         let mut variables = None;
 
         for (key, value) in form_items {
+            // Note: we explicitly decode in the match arms to save work rather
+            // than decoding every form item blindly.
             match key.as_str() {
-                "query" => if query.is_some() {
-                    return Err("Query parameter must not occur more than once".to_owned());
-                } else {
-                    query = Some(value.as_str().to_string());
-                },
-                "operation_name" => if operation_name.is_some() {
-                    return Err(
-                        "Operation name parameter must not occur more than once".to_owned(),
-                    );
-                } else {
-                    operation_name = Some(value.as_str().to_string());
-                },
-                "variables" => if variables.is_some() {
-                    return Err(
-                        "Variables parameter must not occur more than once".to_owned(),
-                    );
-                } else {
-                    variables = Some(serde_json::from_str::<InputValue>(value.as_str())
-                        .map_err(|err| err.description().to_owned())?);
-                },
-                _ => if strict {
-                    return Err(format!("Prohibited extra field '{}'", key).to_owned());
-                },
+                "query" => {
+                    if query.is_some() {
+                        return Err("Query parameter must not occur more than once".to_owned());
+                    } else {
+                        let decoded = value.url_decode().map_err(|_| ()).unwrap();
```

Comment:
We need to properly return an error here instead of unwrapping. 
I also don't get the map_err?